app/config/signupForm.ts

```diff
@@ -1,21 +1,21 @@
-import { FormValues } from '../schemas/userInputSchema';
+import type { UserDataValidation } from '../schemas/userDataValidationSchema';
 import type { TextInputConfig } from '../types/signupForm';
 
 /**
  * Creates a text input configuration object for form fields.
- * @param {keyof FormValues} field - The name of the field.
+ * @param {keyof UserDataValidation} field - The name of the field.
  * @param {string} label - The label text for the field.
- * @param {string} placeholder - The placeholder text for the field.
  * @param {boolean} required - Indicates if the field is required.
+ * @param {string} tooltipLabel - The label text for Tooltip component.
  * @returns {TextInputConfig} The configuration object for a text input field.
  */
 const createConfigItem: (
-  field: keyof FormValues,
+  field: keyof UserDataValidation,
   label: string,
-  placeholder: string,
-  required: boolean
-) => TextInputConfig = (field, label, placeholder, required) => {
-  return { field, label, placeholder, required };
+  required: boolean,
+  tooltipLabel?: string
+) => TextInputConfig = (field, label, required, tooltipLabel) => {
+  return { field, label, required, tooltipLabel };
 };
```

Comment:
Since this function explicitly returns an object of type `TextInputConfig`, we might rename the function to `createTextInputConfig` for clarity.